src/librustc_mir/hair/pattern/check_match.rs

```diff
@@ -527,53 +527,61 @@ fn check_legality_of_move_bindings(cx: &MatchVisitor,
             }
         })
     }
+    let span_vec = &mut Vec::new();
+    {
```

Comment:
I'm not sure I get why you need a new scope here.

Reply:
above point regarding commit, I will ensure I follow that in future :)   Agreed, its best practice to mention everything in the commit, so, if someone backtracks and checks those commit should get the idea of what is being done without going to issue tracker :)

Reply:
@estebank now instead of creating a new scope, I am passing `span_vec` to `check_move` closure, this is a much clearer approach. Thanks for your feedback :)